scripts/importer.py

```diff
@@ -18,60 +18,173 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
-
-
 """Tool to import data from other browsers.
 
 Currently only importing bookmarks from Netscape Bookmark files is supported.
 """
 
-
 import argparse
 
+browser_default_input_format = {
+    'chromium': 'netscape',
+    'ie': 'netscape',
+    'firefox': 'netscape',
+    'seamonkey': 'netscape',
+    'palemoon': 'netscape'
+}
+
 
 def main():
     args = get_args()
-    if args.browser in ['chromium', 'firefox', 'ie']:
-        import_netscape_bookmarks(args.bookmarks, args.bookmark_format)
+    bookmark_types = []
+    output_format = ''
+    input_format = args.input_format
+    if args.search_query or args.search_output:
+        bookmark_types = ['search']
+        if args.oldconfig:
+            output_format = 'oldsearch'
+        else:
+            output_format = 'search'
+    else:
+        if args.bookmark_output:
+            output_format = 'bookmark'
+        elif args.quickmark_output:
+            output_format = 'quickmark'
+        if args.bookmark_query:
+            bookmark_types.append('bookmark')
+        if args.keyword_query:
+            bookmark_types.append('keyword')
+    if not bookmark_types:
+        bookmark_types = ['bookmark', 'keyword']
+    if not output_format:
+        output_format = 'quickmark'
+    if not input_format:
+        input_format = browser_default_input_format[args.browser]
+
+    import_function = {'netscape': import_netscape_bookmarks}
+    import_function[input_format](args.bookmarks, bookmark_types,
+                                  output_format)
 
 
 def get_args():
     """Get the argparse parser."""
     parser = argparse.ArgumentParser(
         epilog="To import bookmarks from Chromium, Firefox or IE, "
-               "export them to HTML in your browsers bookmark manager. "
-               "By default, this script will output in a quickmarks format.")
-    parser.add_argument('browser', help="Which browser? (chromium, firefox)",
-                        choices=['chromium', 'firefox', 'ie'],
-                        metavar='browser')
-    parser.add_argument('-b', help="Output in bookmark format.",
-                        dest='bookmark_format', action='store_true',
-                        default=False, required=False)
+        "export them to HTML in your browsers bookmark manager. "
+        "By default, this script will output in a quickmarks format.")
+    parser.add_argument(
+        'browser',
+        help="Which browser? (chromium, firefox)",
```

Comment:
You probably can delete the part in parens here

Reply:
I've replaced that with a formatter for `choices`, only because the metavar (which makes the usage line look nice) would otherwise hide what browsers are actually supported. `browser` is now an optional argument (given its redundancy if the input format is selected manually). I also cut out the `by default this script will output...` as `-q` already identifies itself as the default in the help text.